### PR TITLE
Strictify //enterprise/app/tap

### PR DESCRIPTION
**Pull request by iain-macdonald**
**Related issues**: N/A

### CODE REVIEW

File: enterprise/app/tap/tap.tsx [hunk 1]
```diff
@@ -403,33 +409,57 @@ export default class TapComponent extends React.Component<Props, State> {
...
+
+    if (!firstStats && !secondStats) {
+      return 0;
+    } else if (!firstStats) {
+      return 1;
+    } else if (!secondStats) {
+      return -1;
+    }
+
+    if (this.getSortMode() == "target") {
```
Comment (jdhollen) [inline enterprise/app/tap/tap.tsx:426]: I don't think this matters a ton, but putting this block (target behavior) before the nullish stats checks would make the behavior more like what it was before, right?

File: enterprise/app/tap/tap.tsx [hunk 2]
```diff
@@ -610,18 +644,18 @@ export default class TapComponent extends React.Component<Props, State> {
...
 
                       let destinationUrl = `/invocation/${status.invocationId}?target=${encodeURIComponent(
-                        targetHistory.target.label
+                        targetHistory.target?.label || ""
                       )}`;
                       let title =
                         this.getColorMode() == "timing"
-                          ? `${this.durationToNum(status.timing.duration).toFixed(2)}s`
-                          : this.statusToString(status.status);
+                          ? `${(this.durationToNum(status.timing?.duration) || 0).toFixed(2)}s`
```
Comment (jdhollen) [inline enterprise/app/tap/tap.tsx:651]: Yeah, could simplify this by making the signature for durationToNum a little cleaner.

File: enterprise/app/tap/tap.tsx [hunk 3]
```diff
@@ -685,15 +719,13 @@ class InnerTopBar extends React.Component<InnerTopBarProps, InnerTopBarState> {
   private hoveredCommitInfo = React.createRef<HTMLDivElement>();
   private hoveredCommitPointer = React.createRef<HTMLDivElement>();
   // TODO(bduffany): Use a generic tooltip component.
-  private tooltipPortal: HTMLDivElement;
+  private tooltipPortal: HTMLDivElement = document.createElement("div");
```
Comment (jdhollen) [inline enterprise/app/tap/tap.tsx:722]: ehhh, rather than doing this outside of the lifecycle, how about just doing ! on the relevant callsites?

**Review by jdhollen — CHANGES_REQUESTED** [review]

File: enterprise/app/tap/tap.tsx [hunk 4]
```diff
@@ -216,36 +216,40 @@ export default class TapComponent extends React.Component<Props, State> {
       let stats: Stat = { count: 0, pass: 0, totalDuration: 0, maxDuration: 0, avgDuration: 0 };
       for (let status of targetHistory.targetStatus) {
         stats.count += 1;
-        let duration = this.durationToNum(status.timing.duration);
+        let duration = this.durationToNum(status.timing?.duration);
```
[the remark this answers is not in the document]
Comment (iain-macdonald) [reply enterprise/app/tap/tap.tsx]: Done

File: enterprise/app/tap/tap.tsx (hunk 1 above)
Comment (iain-macdonald) [reply enterprise/app/tap/tap.tsx:419]: Yup, done

File: enterprise/app/tap/tap.tsx (hunk 2 above)
Comment (iain-macdonald) [reply enterprise/app/tap/tap.tsx]: Done

File: enterprise/app/tap/tap.tsx (hunk 3 above)
Comment (iain-macdonald) [reply enterprise/app/tap/tap.tsx]: Done

**Review by jdhollen — APPROVED** [review]

**Pull request merged**